podcasts/SleepTimerManager.swift

```diff
@@ -0,0 +1,53 @@
+import Foundation
+import PocketCastsUtils
+
+class SleepTimerManager {
+    private var restartSleepTimerIfPlayingAgainWithin: TimeInterval = 5.minutes
+
+    func recordSleepTimerFinished() {
+        Settings.sleepTimerFinishedDate = .now
+    }
+
+    func recordSleepTimerDuration(duration: TimeInterval?, onEpisodeEnd: Bool?) {
+        let setting = SleepTimerSetting(duration: duration, sleepOnEpisodeEnd: onEpisodeEnd)
+        Settings.sleepTimerLastSetting = setting
+    }
+
+    func cancelSleepTimer() {
+        Settings.sleepTimerFinishedDate = .distantPast
+    }
+
+    func restartSleepTimerIfNeeded() {
+        guard !PlaybackManager.shared.sleepTimerActive() else {
```

Comment:
I know it's kinda weird to call `PlaybackManager` here while this is injected there, but I thought it would be better to extract extending sleep timer functionality outside of `PlaybackManager`, given it's already big and complex.